.github/workflows/linux.yml

```diff
@@ -34,23 +38,56 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
 
+    - name: Caching Python dependencies
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
     - name: Install Python packages
       run: |
         python -m pip install numpy setuptools wheel cython pytest pytest-xdist pybind11 pytest-xdist mpmath gmpy2 pythran ninja
         python -m pip install git+https://github.com/rgommers/meson.git@fix-cython-mixed-sources
 
+    - name:  Prepare compiler cache
+      id:    prep-ccache
+      shell: bash
+      run: |
+        mkdir -p "${CCACHE_DIR}"
+        echo "::set-output name=dir::$CCACHE_DIR"
+        NOW=$(date -u +"%F-%T")
+        echo "::set-output name=timestamp::${NOW}"
```

Comment:
Now I'm even more confused. This key will be unique, and will never be matched exactly in a future PR or commit on this branch. Why is this helpful? Is there some ordering of sets of keys so that the most recent timestamp in the past is guaranteed to be found first? If so, it should be explained by a link to the docs or a more detailed comment.

Reply:
https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key does explain it:
```
When a key doesn't match directly, the action searches for keys prefixed with the
restore key. If there are multiple partial matches for a restore key, the action returns
the most recently created cache.
```
@Harsh-Casper if you add a comment with the link and this quote from it, then I think this should indeed work.

Reply:
I have added the same as a detailed comment.

Reply:
That looks good now, thanks.